### PR TITLE
feat:  Allow review rejections to be undone

### PR DESCRIPTION
**Pull request by pselkirk**
Addresses most of #5978 and part of #6169

### CODE REVIEW

**Comment by rjsparks** [comment]
There's at least one rough edge (and this all needs tests).

I ran through having cellar-flac assigned to me, and then rejecting/accepting/rejecting a few times. In my dev instance I've rejected the assignment there are no other assignements, and the open assignments management page doesn't allow the secretary to assign it to me again. The request, as I understand it, is to have me show up on the list here, but marked "has previously rejected reviewing this doc":

image

(right side truncated to not overshare current states)

**Comment by rjsparks** [comment]
@russhousley @ajeanmahoney @kivinen 

This is what the assignment page will look like for a reviewer that has rejected a review before. This is Russ' view after both of us have been assigned and rejected the document.

Is the UI here intuitive enough? And if you have this as secretaries (if you were logged in, Jean, you would see accept buttons for both of us), would you be ok _not_ having the person show up in the list above?

image

**Comment by ajeanmahoney** [comment]
I would typically work from the list on the **Manage unassigned open review requests** page rather than an individual review request page. As secretary, I would expect to see all available reviewers. 

On the individual review request page, the secretary is reassigning to Russ, yes?

Maybe add a date to "Assignment rejected" under the reviewer name, e.g., "Assignment previously rejected 18 Sept 2023". Maybe there should be a **Reject again** button?

File: ietf/review/policies.py [hunk 1]
```diff
@@ -70,6 +69,12 @@ def assign_reviewer(self, review_req, reviewer, add_skip):
         """Assign a reviewer to a request and update policy state accordingly"""
         # Update policy state first - needed by LRU policy to correctly compute whether assignment was in-order
         self.update_policy_state_for_assignment(review_req, reviewer.person, add_skip)
+        assignment = review_req.reviewassignment_set.filter(reviewer=reviewer).first()
+        if assignment:
+            assignment.state = ReviewAssignmentStateName.objects.get(slug='assigned', used=True)
+            assignment.assigned_on = timezone.now()
+            assignment.save()
+            return assignment
         return review_req.reviewassignment_set.create(state_id='assigned', reviewer=reviewer, assigned_on=timezone.now())
```
Comment (rjsparks) [inline ietf/review/policies.py:78]: Consider putting this in an `else:` clause

**Review by rjsparks — APPROVED** [review]

**Pull request merged**